packages/react-components/lib/notifications-dialog.tsx

```diff
@@ -0,0 +1,247 @@
+  DialogActions,
+  DialogContent,
+  IconButton,
+  Paper,
+  Select,
+  MenuItem,
+  Input,
+} from '@material-ui/core';
+import CloseIcon from '@material-ui/icons/Close';
+import moment from 'moment';
+
+export interface Notification {
+  time: string;
+  error: string;
+  severity: string;
+}
+
+export interface NotificationDialogProps {
+  showNotificationsDialog: boolean;
+  setShowNotifications: (payload: boolean) => void;
+  notifications: Notification[];
+}
+
+interface SelectChangeEvent {
+  name?: string | undefined;
+  value: unknown;
+}
+
+interface SeverityIndicatoryProps {
+  severity: string;
+  className?: string;
+}
+
+const useStyles = makeStyles((theme) => ({
+  closeButton: {
+    position: 'absolute',
+    right: theme.spacing(1),
+    top: theme.spacing(1),
+    color: theme.palette.grey[500],
+  },
+  dialogActions: {
+    margin: '0',
+    padding: theme.spacing(1),
+  },
+  dialogContent: {
+    padding: theme.spacing(5),
+  },
+  paper: {
+    display: 'grid',
+    gridTemplateColumns: 'repeat(2, 1fr) 4fr 1fr',
+    textAlign: 'center',
+    padding: theme.spacing(1),
+    width: '100%',
+    margin: `0.5rem 0`,
+  },
+  filter: {
+    display: 'flex',
+    float: 'right',
+    marginBottom: theme.spacing(1),
+  },
+  select: {
+    width: '150px',
+    padding: '0 0.5rem',
+  },
+  removeNotificationIcon: {
+    color: theme.palette.error.main,
+    padding: '0',
+  },
+  legend: {
+    display: 'flex',
+  },
+  indicator: {
+    padding: '0',
+    fontWeight: 600,
+  },
+  placeholder: {
+    color: theme.palette.success.main,
+  },
+}));
+
+const severityStyles = makeStyles((theme) => ({
+  high: {
+    color: theme.palette.secondary.dark,
+  },
+  medium: {
+    color: theme.palette.error.main,
+  },
+  low: {
+    color: theme.palette.warning.light,
+  },
+}));
+
+const SeverityIndicator = (props: SeverityIndicatoryProps): JSX.Element => {
+  const { severity, className } = props;
+  const classes = severityStyles();
+
+  const getStatusLabelClass = (severity: string): string => {
+    switch (severity) {
+      case 'High':
+        return classes.high;
+      case 'Medium':
+        return classes.medium;
+      case 'Low':
+        return classes.low;
+      default:
+        return '';
+    }
+  };
+
+  const styles = `${getStatusLabelClass(severity)} ${className}`;
+  return (
+    <Typography variant="body1" className={styles} align="left">
+      {severity}
+    </Typography>
+  );
+};
+
+export const NotificationsDialog = (props: NotificationDialogProps): JSX.Element => {
+  const classes = useStyles();
+
+  const { showNotificationsDialog, setShowNotifications, notifications } = props;
+
+  const [level, setLevel] = React.useState('');
+  const [rmfNotifications, setRmfNotifications] = React.useState(notifications);
+  // a copy of rmfNotifications
+  // ensure notifications can be restored to correct state after filtering
+  const [notficationsCopy, setNotificationsCopy] = React.useState(notifications);
```

Comment:
Instead of maintaining 3 sets of notification, you can `useMemo` to filter the notifications that should be shown.